shell/browser/api/atom_api_screen.cc

```diff
@@ -99,17 +99,38 @@ static gfx::Rect DIPToScreenRect(electron::NativeWindow* window,
 
 #endif
 
+void DelayEmit(Screen* screen,
+               const base::StringPiece& name,
+               const display::Display& display) {
+  screen->Emit(name, display);
+}
+
+void DelayEmitWithMetrics(Screen* screen,
+                          const base::StringPiece& name,
+                          const display::Display& display,
+                          std::vector<std::string> metrics) {
+  screen->Emit(name, display, metrics);
+}
+
 void Screen::OnDisplayAdded(const display::Display& new_display) {
-  Emit("display-added", new_display);
+  base::PostTaskWithTraits(FROM_HERE, {content::BrowserThread::UI},
+                           base::Bind(&DelayEmit, base::Unretained(this),
```

Comment:
is it ok to use `base::Unretained` here? Is there a guarantee that `Screen` can't be deleted in between posting the task and processing the task?

Reply:
Yes, `Screen` is a singleton that is only destroyed during shutdown.  It won't be ripped out from under itself here afaics

Reply:
inb4 crash on shutdown when kvm switching into a remote desktop or some bs :P

seems fine tho 👍